### DISCUSSION TITLE
SOLWEIG - single point in time via processing tool

### DISCUSSION THREAD

**Discussion by elidamd** [UMEP-dev/UMEP#394] [Q&A] [answered]
Hi!

I am testing SOLWEIG for summer/winter comparisons. While using the processor panel, I noticed that a simulation of one hour is much faster than using an input climate file (with hourly values for a single day). 
I want to use the processing tool mode of SOLWEIG. I prepared an input file with a single row for a given hour, but it returns an error. If I use the climate data file for a day (24 rows), the error does not appear.

Can I use the processing tool or another automated mode for SOLWEIG with single-hour input? Thank you!

UMEP: 3.19.3 
QGIS 3.22.2

Error:
Traceback (most recent call last):
File "C:\Users/elida/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\processing_umep\processor\solweig_algorithm.py", line 642, in processAlgorithm
testwhere = np.where((self.metdata[:, 14] < 0.0) | (self.metdata[:, 14] > 1300.0))
IndexError: too many indices for array: array is 1-dimensional, but 2 were indexed

**Comment by biglimp** [comment, marked as answer]
Yes, try UMEP for processing and see if that solves your issue. Otherwise, try an input file with two lines.

**Reply by elidamd** [reply]
The input with 2 rows worked, it fits with my needs so far. Thanks!
